### PR TITLE
F08 tests use mpi f08

### PR DESCRIPTION
**Pull request by gcorbin**
## Pull Request Description
Some tests in test/mpi/f08 are still implemented with the 'use mpi' interface. Update these tests to use 'use mpi_f08' instead.

* This exposes a bug in the returned indices of MPI_[Wait|Test][any|some] in the 'use mpi_f08' bindings. See #5763 for a fix.
* This PR also udates the F08 tests with fixes from this PR: #3962 

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

File: test/mpi/f08/pt2pt/irsendf08.f90 [hunk 1]
```diff
@@ -78,15 +79,19 @@ subroutine test_pair_irsend( comm, errs )
...
-         index = -1
-         do while (index .ne. 1)
-            call MPI_Waitany(2, requests, index, statuses, ierr)
+         completed = 0
+         do while (completed .lt. 2)
+            call MPI_Waitany(2, requests, index, statuses(1),ierr)
+            if (index .lt. 1 .or. index .gt. 2) then
+                print *, "Invalid index in MPI_Waitany:", index
+            end if
+            if (index .eq. 1) then
```
Comment (gcorbin) [inline test/mpi/f08/pt2pt/irsendf08.f90:88]: The check needs to be inside the loop here to make sure to test the correct request.

**Comment by hzhou** [comment]
test:mpich/ch4/ofi

**Comment by gcorbin** [comment]
> @gcorbin Have you signed [contribution agreement](http://www.mpich.org/documentation/contributor-docs/) yet?

I will submit this as soon as I have the OK from the legal department. Hopefully this won't take too long.

File: test/mpi/f08/pt2pt/irsendf08.f90 (hunk 1 above)
Comment (hzhou) [reply test/mpi/f08/pt2pt/irsendf08.f90:88]: This change can be a separate commit. Could you combine all the other commits into a single commit as they are semantically the same change?

Comment (gcorbin) [reply test/mpi/f08/pt2pt/irsendf08.f90:88]: Done

**Comment by hzhou** [comment]
test:mpich/ch3/most
test:mpich/warnings

File: test/mpi/f08/pt2pt/irsendf08.f90 [hunk 2]
```diff
@@ -25,20 +24,22 @@ program isend
          call test_pair_irsend( comm, errs )
          call mtestFreeComm( comm )
       enddo
-!
+
```
Comment (raffenet) [inline test/mpi/f08/pt2pt/irsendf08.f90:27]: did you mean to remove the `!` at the beginning of this line?

File: test/mpi/f08/datatype/createf90.f90 [hunk 3]
```diff
@@ -1,68 +0,0 @@
...
-        integer ntype1, nsize, ntype2, ntype3, i
-!
-!       Test the Type_create_f90_xxx routines
-!
-        errs = 0
-        call mtest_init( ierr )
-
-! integers with up to 9 are 4 bytes integers; r of 4 are 2 byte,
-! and r of 2 is 1 byte
-        call mpi_type_create_f90_integer( 9, ntype1, ierr )
```
Comment (raffenet) [inline test/mpi/f08/datatype/createf90.f90:22]: there is a F08 binding for this function in the standard. should we update this test to use it?

**Comment by raffenet** [comment]
Hm, #5763 has more commits than this PR. What is the canonical version?

**Comment by hzhou** [comment]
> Hm, #5763 has more commits than this PR. What is the canonical version?

#5763 picked the original commits from this PR. I later suggested the author to squash some commits since most of them do the same thing, just with different tests.

**Comment by raffenet** [comment]
https://github.com/pmodels/mpich/pull/5763/commits/de096a27cff2601458e6c042ec9893ae30422e45 should not have been squashed. Please add it back.

File: test/mpi/f08/datatype/createf90.f90 (hunk 3 above)
Comment (gcorbin) [reply test/mpi/f08/datatype/createf90.f90:22]: There is already an updated version of this test called createf08.f90

Comment (gcorbin) [reply test/mpi/f08/datatype/createf90.f90:22]: There is already an updated version of this test called createf08.f90

File: test/mpi/f08/pt2pt/irsendf08.f90 (hunk 2 above)
Comment (gcorbin) [reply test/mpi/f08/pt2pt/irsendf08.f90:27]: This was an accident

**Comment by gcorbin** [comment]
> [de096a2](https://github.com/pmodels/mpich/commit/de096a27cff2601458e6c042ec9893ae30422e45) should not have been squashed. Please add it back.

I have put the deletion of this file in a separate commit again

**Review by raffenet — APPROVED** [review]
Looks good. Thanks for taking care of this!

**Review by hzhou — APPROVED** [review]
LGTM

**Comment by hzhou** [comment]
test:mpich/warnings

**Pull request merged**